ikarus/io/resultfunction.hh

```diff
@@ -147,45 +153,52 @@ private:
  * \tparam UserFunction Type of the user-defined function for custom result evaluation (default is DefaultUserFunction)
+ * \tparam Args type of additional arguments passed to the user function
+ * \param assembler shared pointer to the underlying assembler (provides the finite elements and the requested results)
+ * \param prec (optional) specify the used precision
+ * \param args additional arguments passed to the constructor of the user function
  */
-template <template <typename, int, int> class RT, typename UserFunction = Impl::DefaultUserFunction, typename AS>
-auto makeResultFunction(std::shared_ptr<AS> assembler, Dune::VTK::Precision prec = Dune::VTK::Precision::float64) {
-  return std::make_shared<ResultFunction<AS, RT, UserFunction>>(assembler, prec);
+template <template <typename, int, int> class RT, typename UserFunction = Impl::DefaultUserFunction,
+          Concepts::FlatAssembler AS, typename... Args>
+auto makeResultFunction(std::shared_ptr<AS> assembler, Dune::VTK::Precision prec = Dune::VTK::Precision::float64,
+                        Args&&... args) {
+  return std::make_shared<ResultFunction<AS, RT, UserFunction>>(assembler, prec, std::forward<Args>(args)...);
 }
 
 /**
  * \brief Function to create a ResultFunction as a gridfunction that can be used with dune-vtk
  * \details
- * Constructs a ResultFunction object with given finite elements, ferequirements as a VTK::Function to be used with
- * dune-vtk It is possible to construct a localFunction from this as follows
+ * Constructs a ResultFunction object with given assembler as a VTK::Function to be used withdune-vtk It is possible to
+ * construct a localFunction from this as follows
  * \code
  * auto localResultFunction = localFunction(vtkResultFunction);
  * localResultFunction.bind(element);
  * \endcode
- * \param assembler shared pointer to the underlying assembler (provides the finite elements and the requested results)
- * \tparam AS  type of the assembler
+ * \tparam AS type of the assembler
  * \tparam RT requested result type
  * \tparam UserFunction Type of the user-defined function for custom result evaluation (default is
  * DefaultUserFunction)
+ * \tparam Args type of additional arguments passed to the user function
+ * \param assembler shared pointer to the underlying assembler (provides the finite elements and the requested results)
+ * \param args additional arguments passed to the constructor of the user function
  */
-template <template <typename, int, int> class RT, typename UserFunction = Impl::DefaultUserFunction, typename AS>
-auto makeResultVtkFunction(std::shared_ptr<AS> assembler) {
-  return Dune::Vtk::Function<typename AS::GridView>(std::make_shared<ResultFunction<AS, RT, UserFunction>>(assembler));
+template <template <typename, int, int> class RT, typename UserFunction = Impl::DefaultUserFunction,
+          Concepts::FlatAssembler AS, typename... Args>
+auto makeResultVtkFunction(std::shared_ptr<AS> assembler, Args&&... args) {
```

Comment:
I suggest that not the args are passed but the final Userfunction. Also in the constructor above.

Then you have the more std lib style and it is less boiler plate here.

Then you can do it like in https://github.com/ikarus-project/ikarus/blob/13b6e5087499abed46bb030cd826b339701ecbe2/ikarus/solver/nonlinearsolver/newtonraphson.hh#L123 and https://github.com/ikarus-project/ikarus/blob/13b6e5087499abed46bb030cd826b339701ecbe2/ikarus/solver/nonlinearsolver/newtonraphson.hh#L101 which is more clean imho.